aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Merger.java

```diff
@@ -109,6 +111,15 @@
                 announceSuccessfulMerge(modelWithMetadata);
```

Comment:
Swallowing `SQLiteConstraintException` errors makes sense, but I think we also need to make sure that we do NOT announce a successful merge.

I think you could do this by doing the `announceSuccessfulMerge` with `andThen`, instead of `doOnComplete`.

It could even make sense to broadcast a new `SUBSCRIPTION_DATA_FAILED` event if a `SQLiteConstraintException` is thrown, but I don't think we need that just yet.

Reply:
order matters! since `Completable` has already encountered error, it has moved onto the `Completable` that was _returned_ by `doOnComplete()`. At this stage, the `SQLiteConstraintException` is caught by `.onErrorComplete()`, which means that `announceSuccessfulMerge()` will not be invoked.

Reply:
Thanks for double checking! Rx is hard :)